config_test.go

```diff
@@ -144,3 +145,72 @@ func TestConfigWithMissingAttributes(t *testing.T) {
 		})
 	}
 }
+
+func makeSamplerCountingHook() (func(zapcore.Entry, zapcore.SamplingDecision),
+	*atomic.Int64, *atomic.Int64) {
+	droppedCount := &atomic.Int64{}
+	sampledCount := &atomic.Int64{}
```

Comment:
nit: use `new` for pointers to zero value.
```suggestion
	droppedCount := new(atomic.Int64)
	sampledCount := new(atomic.Int64)
```

The other convention is `var` and `&` before returning:

```suggestion
	var dropped, sampled atomic.Int64
```

```
  return h, &dropped, &sampled
```